qiskit/passmanager/flow_controllers.py

```diff
@@ -78,8 +77,7 @@ def passes(self) -> list[Task]:
     def iter_tasks(self, state: PassManagerState) -> Generator[Task, PassManagerState, None]:
         max_iteration = self._options.get("max_iteration", 1000)
         for _ in range(max_iteration):
-            for task in self.tasks:
-                state = yield task
+            yield from self.tasks
             if not self.do_while(state.property_set):
                 return
```

Comment:
This is a logic change because the updated `state` isn't received back from the iterator any more.  It might happen to work if the state is only mutated in place, but in general it's not a valid transformation.